Web Inspector: give `WI.StepsFunction` a better name and move it to its own file

dcrousso opened the pull request:
#### 8868d37a317d8eda31bc253ded348505281cb4cc
<pre>
Web Inspector: give `WI.StepsFunction` a better name and move it to its own file
<a href="https://bugs.webkit.org/show_bug.cgi?id=259503">https://bugs.webkit.org/show_bug.cgi?id=259503</a>

Reviewed by Timothy Hatcher and Tim Nguyen.

`WI.StepsFunction` should really be named `WI.StepsTimingFunction` to convey it&apos;s true purpose.

It also shouldn&apos;t really be in `Source/WebInspectorUI/UserInterface/Models/Geometry.js` since it has nothing to do with geometry.

* Source/WebInspectorUI/UserInterface/Models/StepsTimingFunction.js: Added.
(WI.StepsTimingFunction):
(WI.StepsTimingFunction.fromString):
(WI.StepsTimingFunction.prototype.get type):
(WI.StepsTimingFunction.prototype.get count):
(WI.StepsTimingFunction.prototype.copy):
(WI.StepsTimingFunction.prototype.toString):
* Source/WebInspectorUI/UserInterface/Models/Geometry.js:
(WI.Spring):
(WI.StepsFunction): Deleted.
(WI.StepsFunction.fromString): Deleted.
(WI.StepsFunction.prototype.get type): Deleted.
(WI.StepsFunction.prototype.get count): Deleted.
(WI.StepsFunction.prototype.copy): Deleted.
(WI.StepsFunction.prototype.toString): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.prototype._updateEffect):
* Source/WebInspectorUI/UserInterface/Views/AnimationContentView.js:
(WI.AnimationContentView.prototype._refreshPreview):

* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Test.html:

* LayoutTests/inspector/animation/resources/lifecycle-utilities.js:
* LayoutTests/inspector/unit-tests/geometry.html:
* LayoutTests/inspector/unit-tests/geometry-expected.txt:
* LayoutTests/inspector/unit-tests/steps-timing-function.html: Added.
* LayoutTests/inspector/unit-tests/steps-timing-function-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/266345@main">https://commits.webkit.org/266345@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d586c23c6c91abe8a26782cfd635dbbc497179bf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13592 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13906 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14239 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15328 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/12911 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16414 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13986 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15603 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13758 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14400 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11499 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16027 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/11688 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12254 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19299 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12763 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12422 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15636 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12943 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10835 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12213 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/12119 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3310 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16543 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12786 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rcaliman-apple commented on LayoutTests/inspector/unit-tests/steps-timing-function.html:
```diff
@@ -0,0 +1,54 @@
+                InspectorTest.expectEqual(WI.StepsTimingFunction.fromString(inputString).toString(), expectedString, `Should parse '${inputString}' as '${expectedString}'.`);
+            };
+
+            test("steps(1)", "step-end");
+            test("steps(2, jump-start)");
+            test("steps(3, jump-end)");
+            test("steps(4, jump-none)");
+            test("steps(5, jump-both)");
+            test("steps(6, end)");
+            test("steps(7, start)");
+            test("step-start");
+            test("step-end");
```
NIT / personal preference: it was slightly ambiguous which `test()` this refers to because there are multiple declarations of that function name in this file. Nothing bad, but it required me slow down to figure out the scope in my head first. It helped that these were called `testPass()`and `testFail()` in the original file.

rcaliman-apple reviewed (approved):
Looks good to me 👍. Pinging @patrickangle for review and approval.

dcrousso committed:
Web Inspector: give `WI.StepsFunction` a better name and move it to its own file
https://bugs.webkit.org/show_bug.cgi?id=259503

Reviewed by Timothy Hatcher and Tim Nguyen.

`WI.StepsFunction` should really be named `WI.StepsTimingFunction` to convey it's true purpose.

It also shouldn't really be in `Source/WebInspectorUI/UserInterface/Models/Geometry.js` since it has nothing to do with geometry.

* Source/WebInspectorUI/UserInterface/Models/StepsTimingFunction.js: Added.
(WI.StepsTimingFunction):
(WI.StepsTimingFunction.fromString):
(WI.StepsTimingFunction.prototype.get type):
(WI.StepsTimingFunction.prototype.get count):
(WI.StepsTimingFunction.prototype.copy):
(WI.StepsTimingFunction.prototype.toString):
* Source/WebInspectorUI/UserInterface/Models/Geometry.js:
(WI.Spring):
(WI.StepsFunction): Deleted.
(WI.StepsFunction.fromString): Deleted.
(WI.StepsFunction.prototype.get type): Deleted.
(WI.StepsFunction.prototype.get count): Deleted.
(WI.StepsFunction.prototype.copy): Deleted.
(WI.StepsFunction.prototype.toString): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.prototype._updateEffect):
* Source/WebInspectorUI/UserInterface/Views/AnimationContentView.js:
(WI.AnimationContentView.prototype._refreshPreview):

* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Test.html:

* LayoutTests/inspector/animation/resources/lifecycle-utilities.js:
* LayoutTests/inspector/unit-tests/geometry.html:
* LayoutTests/inspector/unit-tests/geometry-expected.txt:
* LayoutTests/inspector/unit-tests/steps-timing-function.html: Added.
* LayoutTests/inspector/unit-tests/steps-timing-function-expected.txt: Added.

Canonical link: https://commits.webkit.org/266345@main
